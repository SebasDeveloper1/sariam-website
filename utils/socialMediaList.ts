import { IconType } from 'react-icons';
import { FaFacebook, FaInstagram } from 'react-icons/fa';

interface SocialMediaListProps {
  title: string;
  icon: IconType;
  url: string;
}

export const socialMediaList: SocialMediaListProps[] = [
  {
    title: 'Facebook',
    icon: FaFacebook,
    url: 'https://www.facebook.com/sariam.party2016',
  },
  {
    title: 'Instagram',
    icon: FaInstagram,
    url: 'https://www.instagram.com/sariam_party/',
  },
];
