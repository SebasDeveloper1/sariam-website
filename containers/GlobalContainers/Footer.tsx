import { createElement, CSSProperties } from 'react';
import Image from 'next/image';
import { Satisfy } from 'next/font/google';
import { socialMediaList } from '@/utils/socialMediaList';
import Logo from '/public/sariam.png';

const satisfy = Satisfy({ subsets: ['latin'], weight: '400' });

export default function Footer() {
  const imageStyle: CSSProperties = {
    objectFit: 'cover',
  };

  const renderSocialMediaIcons = () => {
    return socialMediaList.map((item) => (
      <li key={`social-media-${item.title}`}>
        <a
          href={item.url}
          target="_blank"
          rel="noopener noreferrer"
          title={item.title}
          className="text-4xl text-gray-400 hover:text-gray-500"
        >
          {createElement(item.icon)}
        </a>
      </li>
    ));
  };

  return (
    <section className="w-full py-24 bg-gray-100 border-t-8 border-lime-500">
      <div className="flex flex-col justify-center items-center w-11/12 mx-auto">
        <figcaption className="flex justify-center items-center gap-x-2 mb-8">
          <figure className="relative w-12 aspect-square rounded-full bg-gray-900 border-2 border-lime-500">
            <Image
              src={Logo}
              alt="Logo Sariam Party"
              fill={true}
              style={imageStyle}
              loading="lazy"
              placeholder="blur"
              blurDataURL="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzIwIiBoZWlnaHQ9IjMyMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
            />
          </figure>
          <h3 className={`heading_3 ${satisfy.className} text-gray-900`}>
            SariamParty
          </h3>
        </figcaption>
        <span className="mb-12 font-medium text-center text-gray-500">
          Â© 2023 Sariam Party, Todos los derechos reservados.
        </span>
        <ul className="flex justify-center items-center gap-8 list-none">
          {renderSocialMediaIcons()}
        </ul>
      </div>
    </section>
  );
}
