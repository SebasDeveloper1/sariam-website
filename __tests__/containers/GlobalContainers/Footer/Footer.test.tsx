// Import the required testing utilities from '@testing-library/react'
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

// Import the Footer component and socialMediaList from '@/containers/GlobalContainers/Footer' and '@/utils/socialMediaList' respectively
import Footer from '@/containers/GlobalContainers/Footer';
import { socialMediaList } from '@/utils/socialMediaList';

describe('Footer component', () => {
  // Describe the 'Rendering' section of the Footer component tests
  describe('Rendering', () => {
    // Test whether the Footer component renders correctly
    test('should render Footer component correctly', () => {
      render(<Footer />);
      const footer = document.querySelector('footer');
      expect(footer).toBeInTheDocument();
    });

    // Test whether the logo is rendered correctly with the expected alt text
    test('should render logo correctly', () => {
      const { getByRole } = render(<Footer />);
      const logo = getByRole('img');
      expect(logo).toBeInTheDocument();
      expect(logo).toHaveAttribute('alt', 'Logo Sariam Party');
    });

    // Test whether the title is rendered correctly
    test('should render title correctly', () => {
      const { getByText } = render(<Footer />);
      const title = getByText('SariamParty');
      expect(title).toBeInTheDocument();
    });

    // Test whether the subtitle is rendered correctly
    test('should render subtitle correctly', () => {
      const { getByText } = render(<Footer />);
      const subtitle = getByText(
        'Â© 2023 Sariam Party, Todos los derechos reservados.'
      );
      expect(subtitle).toBeInTheDocument();
    });

    // Test whether the correct number of social media links is rendered
    test('should render social media links correctly', () => {
      const { container } = render(<Footer />);
      const socialMediaLinks = container.querySelectorAll('a');
      expect(socialMediaLinks.length).toBe(socialMediaList.length);
    });
  });

  // Describe the 'Snapshots' section of the Footer component tests
  describe('Snapshots', () => {
    // Test if the rendered output matches the previous snapshot
    test('should match snapshot', () => {
      const { container } = render(<Footer />);
      expect(container).toMatchSnapshot();
    });
  });
});
