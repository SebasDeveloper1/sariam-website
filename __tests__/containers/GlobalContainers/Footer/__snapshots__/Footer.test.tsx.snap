// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Snapshots should match snapshot 1`] = `
<div>
  <footer
    class="relative overflow-hidden w-full py-24 bg-gray-100 border-t-8 border-lime-500"
    role="footer"
  >
    <div
      class="relative flex flex-col justify-center items-center w-11/12 mx-auto"
    >
      <figcaption
        class="flex justify-center items-center gap-x-2 mb-6"
      >
        <figure
          class="relative w-12 aspect-square rounded-full bg-gray-900 border-2 border-lime-500"
        >
          <img
            alt="Logo Sariam Party"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="3rem"
            src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 16w, /_next/image?url=%2Fimg.jpg&w=32&q=75 32w, /_next/image?url=%2Fimg.jpg&w=48&q=75 48w, /_next/image?url=%2Fimg.jpg&w=64&q=75 64w, /_next/image?url=%2Fimg.jpg&w=96&q=75 96w, /_next/image?url=%2Fimg.jpg&w=128&q=75 128w, /_next/image?url=%2Fimg.jpg&w=256&q=75 256w, /_next/image?url=%2Fimg.jpg&w=384&q=75 384w, /_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
          />
        </figure>
        <h3
          class="heading_3 className text-gray-950"
        >
          SariamParty
        </h3>
      </figcaption>
      <span
        class="mb-10 span_base font-medium text-center text-gray-600"
      >
        Â© 2023 Sariam Party, Todos los derechos reservados.
      </span>
      <ul
        class="flex justify-center items-center gap-8 list-none"
      >
        <li
          role="listitem"
        >
          <a
            class="text-4xl text-gray-500 hover:text-gray-600"
            href="https://www.facebook.com/sariam.party2016"
            rel="noopener noreferrer"
            target="_blank"
            title="Facebook"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
              />
            </svg>
          </a>
        </li>
        <li
          role="listitem"
        >
          <a
            class="text-4xl text-gray-500 hover:text-gray-600"
            href="https://www.instagram.com/sariam_party/"
            rel="noopener noreferrer"
            target="_blank"
            title="Instagram"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
              />
            </svg>
          </a>
        </li>
      </ul>
    </div>
  </footer>
</div>
`;
